docs/annotations.rst

```diff
@@ -0,0 +1,108 @@
+Annotations
+===========
+
+SpotBugs supports several annotations to express the developer's intent so that SpotBugs can issue warnings more appropriately.
+Annotations for SpotBugs (mostly deprecated except for SuppressFBWarnings).
+
+CheckForNull
+---------------------------
+The annotated element might be null, and uses of the element should check for null.
+
+CheckReturnValue
+---------------------------
+This annotation is used to denote a method whose return value should always be checked when invoking the method.
+
+CleanupObligation
+---------------------------
+Mark a class or interface as a resource type requiring cleanup.
+
+CreatesObligation
+---------------------------
+Mark a constructor or method as creating a resource which requires cleanup.
+
+DefaultAnnotation
+---------------------------
+Indicates that all members of the class or package should be annotated with the default value of the supplied annotation class.
+
+DefaultAnnotationForFields
+---------------------------
+Indicates that all members of the class or package should be annotated with the default value of the supplied annotation class.
+
+DefaultAnnotationForMethods
+---------------------------
+Indicates that all members of the class or package should be annotated with the default value of the supplied annotation class.
+
+DefaultAnnotationForParameters
+---------------------------
+Indicates that all members of the class or package should be annotated with the default value of the supplied annotation class.
+
+DesireNoWarning (Deprecated)
+---------------------------
+The annotation based approach is useless for lambdas.
+
+DesireWarning (Deprecated)
+---------------------------
+The annotation based approach is useless for lambdas.
+
+DischargesObligation
+---------------------------
+Mark a method as cleaning up a resource.
+
+ExpectWarning (Deprecated)
+---------------------------
+The annotation based approach is useless for lambdas.
+
+NonNull
+---------------------------
+The annotated element must not be null.
+
+NoWarning (Deprecated)
+---------------------------
+The annotation based approach is useless for lambdas.
+
+Nullable
+---------------------------
+The annotated element could be null under some circumstances.
```

Comment:
This should mention that it is treated the same way as not being annotated.

Reply:
Updated based on your feedback.